Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/transport/round_trippers.go

```diff
@@ -130,6 +133,36 @@ func (rt *basicAuthRoundTripper) CancelRequest(req *http.Request) {
 
 func (rt *basicAuthRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt }
 
+type actAsRoundTripper struct {
+	actAs string
+	rt    http.RoundTripper
+}
+
+// NewActAsRoundTripper will apply a BASIC auth authorization header to a
+// request unless it has already been set.
+func NewActAsRoundTripper(actAs string, rt http.RoundTripper) http.RoundTripper {
+	return &actAsRoundTripper{actAs, rt}
+}
+
+func (rt *actAsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if len(req.Header.Get("Act-As")) != 0 {
+		return rt.rt.RoundTrip(req)
+	}
+	req = cloneRequest(req)
+	req.Header.Set("Acting-As", rt.actAs)
```

Comment:
I'm a little weirded out about these being Kubernetes specific.  Can we pick another generic prefix? `X-Authentication-Acting-As`?